pkg/envinfo/storage.go

```diff
@@ -166,21 +208,31 @@ func (ei *EnvInfo) GetStorageMountPath(storageName string) (string, error) {
 		return "", fmt.Errorf("invalid devfile: components.container: required value")
 	}
 
-	// since all container components have same volume mounts, we simply refer to the first container in the list
-	// refer https://github.com/openshift/odo/issues/4105 for addressing "all containers have same volume mounts"
-	paths, err := ei.devfileObj.Data.GetVolumeMountPaths(storageName, containers[0].Name)
-	if err != nil {
-		return "", err
+	var paths, uniquePaths []string
+
+	// go over all containers
+	for _, c := range containers {
+		// get all volume mount paths in current container
+		pt, err := ei.devfileObj.Data.GetVolumeMountPaths(storageName, c.Name)
+		if err != nil {
+			return "", err
+		}
+		paths = append(paths, pt...)
```

Comment:
Since we care about only a single path for now, we can return from here and leave a comment for later.

Reply:
Done